specification/ai/Azure.AI.Projects/main.tsp

```diff
@@ -0,0 +1,65 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "./agents/routes.tsp";
+import "./agents/threads/routes.tsp";
+import "./agents/messages/routes.tsp";
+import "./agents/runs/routes.tsp";
+import "./agents/run_steps/routes.tsp";
+import "./agents/files/routes.tsp";
+import "./agents/streaming/events.tsp";
+import "./agents/vector_stores/routes.tsp";
+import "./agents/vector_stores/files/routes.tsp";
+import "./agents/vector_stores/file_batches/routes.tsp";
+import "./connections/routes.tsp";
+import "./evaluations/routes.tsp";
+import "./datasets/routes.tsp";
+import "./indexes/routes.tsp";
+import "./deployments/routes.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Foundations;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+
+#suppress "@azure-tools/typespec-azure-core/casing-style"
+namespace Azure.AI {
+
+}
+
+@useAuth(
+  OAuth2Auth<[
```

Comment:
please also support api-key auth via the AUTHORIZATION header

Reply:
I'll make a TypeSpec change to support api-key as soon as we know which HTTP request header to use.